src/transaction.cpp

```diff
@@ -138,6 +139,7 @@ int wsrep::transaction::start_transaction(
     state_hist_.clear();
     ws_handle_ = wsrep::ws_handle(id);
     flags(wsrep::provider::flag::start_transaction);
+    client_service_.notify_state_change(wsrep_hrtime::hrtime());
```

Comment:
Why `hrtime()` needs to be passed as an argument to `notify_state_change()`? This fixes the timestamp type and resolution to what `hrtime()` returns. The (server side) implementatin of `notify_state_change()` could get the timestamp as well, and in that case the wsrep-lib would not need to define the timestamp type and resolution.

Reply:
You are correct that server side could do it as well but some time would have passed from actual state change, not sure if it is relevant?

Reply:
The difference in timing is not significant, it is probably much smaller than the time resolution of `hrtime()` (a few instructions).

Reply:
Sure, I will move this to server code.